### ISSUE TITLE
[BUG]: Pipeline_name X not unique within workspace

### ISSUE THREAD

**Issue by MingStar** [zenml-io/zenml#1362]
### System Information

ZENML_LOCAL_VERSION: 0.34.0
ZENML_SERVER_VERSION: 0.34.0
ZENML_SERVER_DATABASE: sqlite
ZENML_SERVER_DEPLOYMENT_TYPE: other
ZENML_CONFIG_DIR: C:\Users\weiming.chen\AppData\Roaming\zenml
ZENML_LOCAL_STORE_DIR: C:\Users\weiming.chen\AppData\Roaming\zenml\local_stores
ZENML_SERVER_URL: sqlite:///C:\Users\weiming.chen\AppData\Roaming\zenml\local_stores\default_zen_store/zenml.db
ZENML_ACTIVE_REPOSITORY_ROOT: C:\Users\weiming.chen\Code\spark_bulk_data_loader
PYTHON_VERSION: 3.10.10
ENVIRONMENT: native
SYSTEM_INFO: {'os': 'windows', 'windows_version_release': '10', 'windows_version': '10.0.19044', 'windows_version_service_pack': 
'SP0', 'windows_version_os_type': 'Multiprocessor Free'}
ACTIVE_WORKSPACE: default
ACTIVE_STACK: local_ceph
ACTIVE_USER: default
TELEMETRY_STATUS: disabled
ANALYTICS_CLIENT_ID: 2de1bc7d-7606-4976-b974-59da17a9a573
ANALYTICS_USER_ID: fed4dcbe-35f6-4de5-b01f-5cdb39665aef
ANALYTICS_SERVER_ID: 2de1bc7d-7606-4976-b974-59da17a9a573
INTEGRATIONS: ['deepchecks', 'evidently', 'kaniko', 'mlflow', 'pillow', 'plotly', 'pytorch', 's3', 'scipy', 'sklearn']
PACKAGES: {'tifffile': '2023.1.23.1', 'certifi': '2022.12.7', 'fsspec': '2022.11.0', 's3fs': '2022.11.0', 'pytz': '2022.7.1',      
'pytzdata': '2020.1', 'pywin32': '305', 'setuptools': '65.0.0', 'cryptography': '39.0.1', 'pyzmq': '25.0.0', 'packaging': '23.0',  
'pip': '23.0', 'gevent': '22.10.2', 'attrs': '22.2.0', 'aiofiles': '22.1.0', 'argon2-cffi': '21.3.0', 'argon2-cffi-bindings':      
'21.2.0', 'virtualenv': '20.16.3', 'isoduration': '20.11.0', 'rich': '12.6.0', 'pyarrow': '11.0.0', 'websockets': '10.4', 'pillow':
'9.4.0', 'click': '8.1.3', 'tenacity': '8.1.0', 'jupyter-client': '8.0.3', 'ipython': '7.34.0', 'ipywidgets': '7.7.3', 'nbconvert':
'7.2.9', 'pytest': '7.2.1', 'coverage': '7.2.0', 'ipykernel': '6.21.2', 'notebook': '6.5.2', 'tornado': '6.2', 'multidict':        
'6.0.4', 'docker': '6.0.1', 'bleach': '6.0.0', 'importlib-metadata': '6.0.0', 'psutil': '5.9.4', 'traitlets': '5.9.0', 'nbformat': 
'5.7.3', 'zope.interface': '5.5.2', 'plotly': '5.5.0', 'pyyaml': '5.4.1', 'jupyter-core': '5.2.0', 'decorator': '5.1.1', 'smmap':  
'5.0.0', 'tqdm': '4.64.1', 'fonttools': '4.38.0', 'jsonschema': '4.17.3', 'beautifulsoup4': '4.11.2', 'rsa': '4.9', 
'opencv-python-headless': '4.7.0.68', 'zope.event': '4.6', 'opencv-python': '4.5.5.64', 'typing-extensions': '4.5.0', 'imagehash': 
'4.3.1', 'gitdb': '4.0.10', 'async-timeout': '4.0.2', 'bcrypt': '4.0.1', 'pytest-cov': '4.0.0', 'protobuf': '3.20.3', 'orjson':    
'3.8.5', 'aiohttp': '3.8.3', 'zipp': '3.8.1', 'filelock': '3.8.0', 'matplotlib': '3.7.0', 'anyio': '3.6.2', 'widgetsnbextension': 
'3.6.2', 'asgiref': '3.6.0', 'python-utils': '3.4.5', 'idna': '3.4', 'cfgv': '3.3.1', 'python-jose': '3.3.0', 'oauthlib': '3.2.2', 
'gitpython': '3.1.31', 'jinja2': '3.1.2', 'pyspark': '3.1.2', 'threadpoolctl': '3.1.0', 'prompt-toolkit': '3.0.37', 'jsonpickle':  
'3.0.1', 'networkx': '3.0', 'platformdirs': '3.0.0', 'tritonclient': '2.29.0', 'requests': '2.28.2', 'imageio': '2.25.0', 
'pycparser': '2.21', 'pre-commit': '2.20.0', 'fastjsonschema': '2.16.2', 'pygments': '2.14.0', 'python-dateutil': '2.8.2',         
'category-encoders': '2.6.0', 'pyjwt': '2.6.0', 'identify': '2.5.3', 'pyparsing': '2.4.7', 'aiobotocore': '2.4.2', 'soupsieve':    
'2.4', 'jsonpointer': '2.3', 'jupyter-server': '2.3.0', 'flask': '2.2.2', 'werkzeug': '2.2.2', 'asttokens': '2.2.1', 'cloudpickle':
'2.2.1', 'itsdangerous': '2.1.2', 'markupsafe': '2.1.2', 'pendulum': '2.1.2', 'waitress': '2.1.2', 'charset-normalizer': '2.1.1',  
'wtg-pyspark-shared': '2.1.0', 'pywinpty': '2.0.10', 'python-json-logger': '2.0.7', 'mistune': '2.0.5', 'geventhttpclient':        
'2.0.2', 'greenlet': '2.0.2', 'kafka-python': '2.0.2', 'shapely': '2.0.1', 'tomli': '2.0.1', 'iniconfig': '2.0.0', 'grpcio':       
'1.51.1', 'botocore': '1.27.59', 'urllib3': '1.26.14', 'mlflow': '1.25.1', 'boto3': '1.24.59', 'numpy': '1.24.2', 'six': '1.16.0', 
'cffi': '1.15.1', 'wrapt': '1.14.1', 'torch': '1.13.0', 'webcolors': '1.12', 'py': '1.11.0', 'pydantic': '1.10.5', 'scipy':        
'1.10.1', 'backoff': '1.10.0', 'python-rapidjson': '1.9', 'yarl': '1.8.2', 'alembic': '1.8.1', 'distro': '1.8.0', 'send2trash':    
'1.8.0', 'passlib': '1.7.4', 'nodeenv': '1.7.0', 'debugpy': '1.6.6', 'monotonic': '1.6', 'nest-asyncio': '1.5.6', 'pandas':        
'1.5.3', 'fqdn': '1.5.1', 'websocket-client': '1.5.1', 'pandocfilters': '1.5.0', 'sqlalchemy': '1.4.41', 'kiwisolver': '1.4.4',    
'atomicwrites': '1.4.1', 'jpype1': '1.4.1', 'pywavelets': '1.4.1', 'analytics-python': '1.4.0', 'frozenlist': '1.3.3', 'aiosignal':
'1.3.1', 'albumentations': '1.3.0', 'sniffio': '1.3.0', 'mako': '1.2.4', 'querystring-parser': '1.2.4', 'arrow': '1.2.3', 
'jaydebeapi': '1.2.3', 'mlserver': '1.2.3', 'mlserver-mlflow': '1.2.3', 'scikit-learn': '1.2.1', 'tinycss2': '1.2.1', 'executing': 
'1.2.0', 'joblib': '1.2.0', 'uri-template': '1.2.0', 'jupyterlab-widgets': '1.1.2', 'exceptiongroup': '1.1.0', 'brotli': '1.0.9',  
'contourpy': '1.0.7', 'pymysql': '1.0.2', 'jmespath': '1.0.1', 'pluggy': '1.0.0', 'fastapi': '0.75.2', 'sqlalchemy-utils':         
'0.38.3', 'wheel': '0.37.1', 'zenml': '0.34.0', 'python-dotenv': '0.21.1', 'prometheus-flask-exporter': '0.21.0', 'pyrsistent':    
'0.19.3', 'scikit-image': '0.19.3', 'httplib2': '0.19.1', 'jedi': '0.18.2', 'validators': '0.18.2', 'ecdsa': '0.18.0', 'uvicorn':  
'0.17.6', 'databricks-cli': '0.17.4', 'starlette': '0.17.1', 'terminado': '0.17.1', 'prometheus-client': '0.16.0', 
'starlette-exporter': '0.15.1', 'h11': '0.14.0', 'torchvision': '0.14.0', 'statsmodels': '0.13.5', 'seaborn': '0.12.2', 
'aioitertools': '0.11.0', 'cycler': '0.11.0', 'py4j': '0.10.9', 'toml': '0.10.2', 'python-terraform': '0.10.1', 'commonmark':      
'0.9.1', 'tabulate': '0.9.0', 'parso': '0.8.3', 'watchgod': '0.8.2', 'aiokafka': '0.8.0', 'deepchecks': '0.8.0', 'pickleshare':    
'0.7.5', 'nbclient': '0.7.2', 'defusedxml': '0.7.1', 'py-grpc-prometheus': '0.7.0', 'jupyter-events': '0.6.3', 'stack-data':       
'0.6.2', 'dataclasses': '0.6', 's3transfer': '0.6.0', 'patsy': '0.5.3', 'nbclassic': '0.5.2', 'webencodings': '0.5.1', 'httptools':
'0.5.0', 'pytorch-ignite': '0.4.10', 'pyasn1': '0.4.8', 'colorama': '0.4.6', 'jupyter-server-terminals': '0.4.4', 'sqlparse':      
'0.4.3', 'entrypoints': '0.4', 'imgaug': '0.4.0', 'distlib': '0.3.5', 'pynomaly': '0.3.3', 'click-params': '0.3.0', 'wcwidth':     
'0.2.6', 'jupyterlab-pygments': '0.2.2', 'notebook-shim': '0.2.2', 'pure-eval': '0.2.2', 'fastapi-utils': '0.2.1', 'backcall':     
'0.2.0', 'ipython-genutils': '0.2.0', 'evidently': '0.1.54.dev0', 'matplotlib-inline': '0.1.6', 'rfc3339-validator': '0.1.4',      
'comm': '0.1.2', 'rfc3986-validator': '0.1.1', 'sqlmodel': '0.0.8', 'python-multipart': '0.0.5', 'qudida': '0.0.4', 
'sqlalchemy2-stubs': '0.0.2a32'}

### What happened?

started using 0.34.0 with versioning of the pipelines, therefore I had multiple versions of a pipeline with the same name.

however, when I called `pipeline_instance.get_runs()[-1]` to try to retrieve the last run information, it throws a RuntimeError:

``Pipeline_name `X` not unique within workspace ``

It seems the helper function `get_pipeline()` in `post_execution/pipeline.py` has not taken into account of the version info or trying to get the latest version of a pipeline model

### Reproduction steps

1.
2.
3.
...

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by schustmi** [comment]
Should be fixed in #1363
